session/common.go

```
@@ -330,13 +330,14 @@ func (x *commonData) SetIssuer(id user.ID) {
 	x.issuer = id
 }
 
-// AssertAuthKey asserts public key bound to the session.
+// AssertAuthKey asserts public key bound to the session. The key should not be
+// nil.
```

If it shouldn't be nil, than why should we check for nilness below?

---

u think we can let it panic?

---

We can consider it a misuse of interface and just not handle this in any way (it's the default assumption for any interface accepting a pointer). Or if for whatever reason we have a lot of cases where passing a `nil` here would be convenient we can explicitly accept it and return false (which is usually documented specifically).

---

I just don't know where you've encountered a case requiring this patch.

---

in mentioned unit test. No app passes nil, and none should. This is obvious, so i think that it is unnecessary even in docs